src/net/socket.rs

```diff
@@ -103,17 +120,18 @@ impl Socket {
     }
 
     // Serializes and sends a `Packet` on the socket. On success, returns the number of bytes written.
-    fn send_to(&mut self, packet: Packet) -> Result<usize> {
-        let connection = self
-            .connections
-            .get_or_insert_connection(packet.addr(), &self.config);
+    fn send_to(&mut self, time: Instant, packet: Packet) -> Result<usize> {
+        let connection =
+            self.connections
+                .get_or_insert_connection(packet.addr(), time, &self.config);
```

Comment:
If we do it this way, we will be actively losing state of the first few messages a remote client has sent us because we're creating a new VirtualConnection here. Put in a different way, we won't be able to ack the first few messages which could cause problems.

Reply:
Should we ack the first message? Why not let the client retry sending a few times instead? If we add the connection to the connection table we get the DoS problem again.